Do not always install the dracut module in hostonly mode

aafeijoo-suse opened the pull request:
For hostonly initrds, check that the system has a TPM2 device. This check does not affect generic initrds.

aplanas commented on module-setup.sh:
```diff
@@ -3,6 +3,12 @@
 # Prerequisite check(s) for module.
 check() {
     # Return 255 to only include the module, if another module requires it.
+    if [[ $hostonly ]]; then
```
Can you use `[ -n "$hostonly" ]`?

For some reason, using  `[` instead of `[[` is the idiomatic way in this code base.

aafeijoo-suse replied:
Within the dracut code base, this check is always performed with `[[ $hostonly ]]`, but ok, I can change it anyway :)

aplanas commented on module-setup.sh:
```diff
@@ -3,6 +3,12 @@
 check() {
     # Return 255 to only include the module, if another module requires it.
+    if [[ $hostonly ]]; then
+        if ! [[ -d /sys/class/tpmrm ]] || [[ -z "$(ls -A /sys/class/tpmrm)" ]]; then
```
why `! [ -d /sys/class/tpmrm ]` alone is not enough?

aafeijoo-suse replied:
Check [1] or [2] for a reference implementation in systemd ([2] is what sdbootutil's `have_tpm2()` calls internally).

[1] https://github.com/systemd/systemd/blob/fea7f5d68b5a6272ce8989e872a57b42121b5205/src/shared/tpm2-util.c#L8761
[2] https://github.com/systemd/systemd/blob/07f471824288433d17ebe097e94dfca85e74fea6/src/shared/tpm2-util.c#L6268

aplanas replied:
Ah I see! Thanks!

aplanas reviewed (commented):
LGTM, some comments

aafeijoo-suse commented:
FTR, I came across this while testing a vm with grub2-bls without TPM2, where sdbootutil requires dracut-pcr-signature, and both packages install dracut modules related to TPM2/PCRs. Avoiding both dracut modules when they are not needed reduces the size of the initrd by ~1 MB.

aplanas commented on module-setup.sh:
```diff
@@ -3,6 +3,12 @@
 check() {
     # Return 255 to only include the module, if another module requires it.
+    if [ -n "$hostonly" ]; then
+        if ! [[ -d /sys/class/tpmrm ]] || [[ -z "$(ls -A /sys/class/tpmrm)" ]]; then
```
please, `[` instead of `[[`?

aafeijoo-suse replied:
Sorry, I thought that you just wanted me to change only the condition without operator. Done.

aafeijoo-suse committed:
Do not always install the dracut module in hostonly mode

For hostonly initrds, check that the system has a TPM2 device. This check does
not affect generic initrds.
